### DISCUSSION TITLE
cascading _count ?

### DISCUSSION THREAD

**Discussion by TamirCode** [prisma/prisma#19857] [Q&A] [answered]
A post has comments, and a comments has replies (comments) 
when I count comments of a post, it only has first layer of comments. What should I do to include the rest of the nested comments in the count?
```js
prisma.post.findMany({ include: { _count: { select: { comments: true } } } }))
```

```prisma
model User {
    id        Int      @id @default(autoincrement())
    username  String   @unique
    password  String
    settings  Json?
    role      Role     @default(USER)
    createdAt DateTime @default(now())
    updatedAt DateTime @updatedAt

    // one to many
    posts    Post[]    @relation("posts")
    comments Comment[] @relation("comments")
}

model Post {
    id        Int      @id @default(autoincrement())
    content   String
    createdAt DateTime @default(now())
    updatedAt DateTime @updatedAt

    // many to one
    user     User      @relation("posts", fields: [userId], references: [id])
    userId   Int
    comments Comment[]
}

model Comment {
    id        Int      @id @default(autoincrement())
    content   String
    createdAt DateTime @default(now())
    updatedAt DateTime @updatedAt

    // one to many
    user   User  @relation("comments", fields: [userId], references: [id])
    userId Int
    post   Post? @relation(fields: [postId], references: [id])
    postId Int?

    // one to many self-reference
    childComments   Comment[] @relation("replies")
    // many to one self-reference
    parentComment   Comment?  @relation("replies", fields: [parentCommentId], references: [id])
    parentCommentId Int?
}
```

**Comment by ludralph** [comment]
Hi @TamirCode 👋 

Can you please clarify if there will be more than one level of nesting in the comments?  Can a child comment also have a nested comment?

**Reply by TamirCode** [reply]
Yes that is the case. A comment can have nested comments and so on.
I am open to alternative ways to structure the models if needed

**Reply by ludralph** [reply, marked as answer]
Hi @TamirCode, in this case, you may want to go with a recursive approach since you want to query to an arbitrary depth or until you reach the end, Prisma doesn't support that yet. Take a look at this feature request and feel free to add your +1/use case.

- https://github.com/prisma/prisma/issues/3725

You can also take a look at this `filtered relation count aggregate` feature request which looks like what you want to achieve

- https://github.com/prisma/prisma/issues/6570

Depending on your database provider, you might be able to query recursively using the [`prisma.$queryRaw`](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw) syntax. [Here's an article](https://www.postgresqltutorial.com/postgresql-recursive-query/) about doing so with Postgres. You can also leverage [SafeQL](https://safeql.dev/compatibility/prisma.html) for building your raw queries. 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
Please let us know in case you have any further queries!

**Reply by nurul3101** [reply]
Thanks for marking this as answered and resolving the discussion 🙏
